src/backend/cuda/subconfigure.m4

```diff
@@ -41,3 +77,121 @@ elif test "${enable_cuda_p2p}" = "cliques" ; then
 else
     AC_DEFINE([CUDA_P2P],[CUDA_P2P_DISABLED],[Define if CUDA P2P is disabled])
 fi
+
+
+##########################################################################
+##### analyze the user arguments and setup internal infrastructure
+##########################################################################
+
+if test ${have_cuda} = "yes" ; then
+    for maj_version in 11 10 9 8 7 6 5 ; do
+        version=$((maj_version * 1000))
+        AC_COMPILE_IFELSE([AC_LANG_PROGRAM([
+                              #include <cuda.h>
+                              int x[[CUDA_VERSION - $version]];
+                          ],)],[cuda_version=${maj_version}],[])
+        if test ! -z ${cuda_version} ; then break ; fi
+    done
+    PAC_PUSH_FLAG([IFS])
+    IFS=","
+    CUDA_SM=
+    for sm in ${with_cuda_sm} ; do
+        case "$sm" in
+            all)
+                if test ${cuda_version} -ge 11 ; then
+                    # ampere
+                    PAC_APPEND_FLAG([80],[CUDA_SM])
+                fi
+
+                if test ${cuda_version} -ge 10 ; then
+                    # turing
+                    PAC_APPEND_FLAG([75],[CUDA_SM])
+                fi
+
+                if test ${cuda_version} -ge 9 ; then
+                    # volta
+                    PAC_APPEND_FLAG([70],[CUDA_SM])
+                    PAC_APPEND_FLAG([72],[CUDA_SM])
+                fi
+
+                if test ${cuda_version} -ge 8 ; then
+                    # pascal
+                    PAC_APPEND_FLAG([60],[CUDA_SM])
+                    PAC_APPEND_FLAG([61],[CUDA_SM])
+                    PAC_APPEND_FLAG([62],[CUDA_SM])
+                fi
+
+                if test ${cuda_version} -ge 6 ; then
+                    # maxwell
+                    PAC_APPEND_FLAG([50],[CUDA_SM])
+                    PAC_APPEND_FLAG([52],[CUDA_SM])
+                    PAC_APPEND_FLAG([53],[CUDA_SM])
+                fi
+
+                if test ${cuda_version} -ge 6 ; then
+                    # kepler
+                    PAC_APPEND_FLAG([30],[CUDA_SM])
+                    PAC_APPEND_FLAG([35],[CUDA_SM])
+                    PAC_APPEND_FLAG([37],[CUDA_SM])
+                fi
+                ;;
```

Comment:
Nit pick: one branch

Reply:
Yikes, it should be `-ge 5` for `kepler`.  Thanks for catching it.  I'll fix it.